### ISSUE TITLE
Autohost message again

### ISSUE THREAD

**Issue by D3thw0lf** [TwitchLib/TwitchLib#326]
I have to reopen a similar issue related to that one here I opened ages ago ( #220 )

I tried the following code:
```c
private async void OnGetHosted(object sender, OnBeingHostedArgs e)
{
   if (e.IsAutoHosted)
   {
     MessageBox.Show($"{e.HostedByChannel} is Autohosting you with {e.Viewers} Viewers.");
   }
}
```
/edit before you ask, yes I added this ofc
```c
newClient.OnBeingHosted += OnGetHosted;
```

Maybe I'm doing the request wrong but it seems like I can't get it to work (no response, no box at all it just not triggers) normal hosting works (so if someone is hosting per /host <channel> command)

Would be great to hear from you what I'm doing wrong (again) :D and as always thanks for your help in advance. 👍 Hope you had a great Xmas (or if you read that in earlier 2018) a great start into the new year.

**Comment by swiftyspiffy** [comment]
So the TwitchClient fires that event when it receives a message from Twitch similar to:
```
BurkeBlack is now hosting you for up to 206 viewers.
```
Is this message still in use by Twitch, or has it been changed/removed? I don't know, but I'll start looking for it when I'm in Twitch chat.

**Comment by D3thw0lf** [comment]
Yup the normal hosting (and raids) works just fine. It just not fires the autohosted one. Thanks and just take the time it needs ... just wanted to know if I'm doing something wrong or maybe it's a more general issue. :)

**Comment by swiftyspiffy** [comment]
So I just tried to host a (offline) channel,  and I didn't get a message in the hosted channel. Perhaps I'm doing it wrong. Are you able to get the IRC string for this kind of event? I'll keep trying.

**Comment by D3thw0lf** [comment]
I guess the event on twitch only fires if the channel is live maybe that's why you won't get any notification. ;)

In this thread (I mentioned in #220) they said, they added "is auto hosting" if the source is from an autohost (https://discuss.dev.twitch.tv/t/differentiate-host-and-auto-host/7829/6)

So I guess it's
```
:jtv!jtv@jtv.tmi.twitch.tv PRIVMSG annemunition :WhateverChannelNameHere is auto hosting you for up to 100 viewers.
```
or similar

**Comment by swiftyspiffy** [comment]
I meant to say online channel in the edit. Perhaps it's only visible to the broadcaster?

**Comment by D3thw0lf** [comment]
Yup it's only visible as channel owner. It looks like these "your friend xyz started to watch, too" messages

**Comment by swiftyspiffy** [comment]
Sorry for the delay on this, if you want to give it a go now, it should work as expected. OnBeingHosted has a IsAutoHost bool that allows you to differentiate between manual and auto hosts.

**Comment by D3thw0lf** [comment]
Ah nice ... I'll try this :)

**Comment by D3thw0lf** [comment]
So I finally could try this in a livetest. For some reason they don't trigger yet. I used the same code as above. :(

But I've another issue which is maybe on my end but for some reason my client closes if (for example) the follow event gets casted. But I'll take a close look on my stuff first before I open up an issue for that. :)

Just wanted to let you know: Autohost is not yet triggering for some reasons I don't know. Maybe Twitch has changed the way they get displayed/used? For some reason I don't even get those "xyz is autohosting for X viewers" message on their chat similar to that "hey your friend xyz is now watching" message anymore.

**Comment by swiftyspiffy** [comment]
Odd, I’ll listen to a chat and watch for hosting messages. The only reason the current code wouldn’t work is because either you’re not receiving the host messages at all (which you indicated), or because the message(s) were altered beyond what I used to write he parse code.

**Comment by D3thw0lf** [comment]
Found the issue why it crashes btw. :)
```
Unbehandelte Ausnahme: System.FormatException: Die Eingabezeichenfolge hat das falsche Format.
   bei System.Number.StringToNumber(String str, NumberStyles options, NumberBuffer& number, NumberFormatInfo info, Boolean parseDecimal)
   bei System.Number.ParseInt32(String s, NumberStyles style, NumberFormatInfo info)
   bei System.Int32.Parse(String s)
   bei TwitchLib.TwitchClient.ParseIrcMessage(String ircMessage) in C:\Users\max45\Documents\Visual Studio 2017\Projects\TwitchLib\TwitchLib\TwitchClient.cs:Zeile 849.
   bei TwitchLib.TwitchClient._client_OnMessage(Object sender, MessageReceivedEventArgs e) in C:\Users\max45\Documents\Visual Studio 2017\Projects\TwitchLib\TwitchLib\TwitchClient.cs:Zeile 571.
   bei WebSocket4Net.WebSocket.FireMessageReceived(String message)
   bei WebSocket4Net.Command.Text.ExecuteCommand(WebSocket session, WebSocketCommandInfo commandInfo)
   bei WebSocket4Net.WebSocket.OnDataReceived(Byte[] data, Int32 offset, Int32 length)
   bei WebSocket4Net.WebSocket.client_DataReceived(Object sender, DataEventArgs e)
   bei SuperSocket.ClientEngine.ClientSession.OnDataReceived(Byte[] data, Int32 offset, Int32 length)
   bei SuperSocket.ClientEngine.AsyncTcpSession.ProcessReceive(SocketAsyncEventArgs e)
   bei SuperSocket.ClientEngine.AsyncTcpSession.SocketEventArgsCompleted(Object sender, SocketAsyncEventArgs e)
   bei System.Net.Sockets.SocketAsyncEventArgs.OnCompleted(SocketAsyncEventArgs e)
   bei System.Net.Sockets.SocketAsyncEventArgs.ExecutionCallback(Object ignored)
   bei System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   bei System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   bei System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   bei System.Net.Sockets.SocketAsyncEventArgs.FinishOperationSuccess(SocketError socketError, Int32 bytesTransferred, SocketFlags flags)
   bei System.Net.Sockets.SocketAsyncEventArgs.CompletionPortCallback(UInt32 errorCode, UInt32 numBytes, NativeOverlapped* nativeOverlapped)
   bei System.Threading._IOCompletionCallback.PerformIOCompletionCallback(UInt32 errorCode, UInt32 numBytes, NativeOverlapped* pOVERLAP)
Ausnahme ausgelöst: "System.Net.Sockets.SocketException" in System.dll
Ausnahme ausgelöst: "System.Net.Sockets.SocketException" in System.dll
```

**Comment by swiftyspiffy** [comment]
Do you have the string sent by Twitch that was being parsed? That’d be really useful.

**Comment by D3thw0lf** [comment]
So ... new information: Autohost is not the problem (it's not casting but that's not why it crashes). I've screenshotet this for you (hope it helps to resolve the issue).

![crash](https://user-images.githubusercontent.com/895448/35158944-ebc23070-fd38-11e7-9869-035402e0c51d.jpg)

**Comment by D3thw0lf** [comment]
So ... I've tested with a friend, he hosted with me in his channel so I was definitly there as at least +1 viewer but the message still is
```
:jtv!jtv@jtv.tmi.twitch.tv PRIVMSG d3thw0lf :Amarouk_Rex is now hosting you.
```
No viewercount anymore in that kind of message as it seems. :(

**Comment by swiftyspiffy** [comment]
Shit, I totally forgot that it's possible to host someone with no viewers (I think 1 viewer or less). I'll make a change for that here shortly.

**Comment by D3thw0lf** [comment]
Thanks Cole,

I'll check that asap I can grab your fix. 👍 :)

And ofc thanks for your hard work my friend. ^w^

**Comment by swiftyspiffy** [comment]
Go ahead and give this a go again, it should work for both messages with a number of viewers, and with none.

**Comment by D3thw0lf** [comment]
Host Alert works again (with 0 and more Viewers) so first things first: thanks for fixing that! 👍 
Autohost sadly still don't trigger any alert/messagebox etc. (for me) ... I hope this is not a "regional thing" at twitch (would be rly sad) so maybe someone tried this, too and can respond to this issue here.

I'll rewrite my code (at the beginning of this issue) to make sure it's rly not something I missed and re-check completly but for now I can't get this to work ... trying more stuff and coming back then. :)

In short: Host stuff works again, Autohost still doesn't :3

**Comment by swiftyspiffy** [comment]
Hmm. It should fire based on the provided irc snippets above. Are you certain you are receiving the snippets above? Does the log show that they’re being received?

**Comment by D3thw0lf** [comment]
I checked my logfiles after my last +4h Stream and I clearly not getting that autohost messages for ... reasons(?)

Could it be possible that not all IRC Servers of Twitch (yet) showing this message? I'm very sure I had those messages after they released this feature and ofc I can see them in the Dashboard (under the "Autohosts you"-Tab).
![autohostsme](https://user-images.githubusercontent.com/895448/35735268-6063d94c-0824-11e8-9733-155aad1f26b6.jpg)

**Comment by D3thw0lf** [comment]
Update: It seems like to work now. I've got an "Is autohosting you" Message. Maybe Twitch finally changed something or fixed a missing IRC Message. Thanks for your hard Work. I'll monitor this particular Issue for a bit longer but I guess you can add it to the "fixed" list. 👍

**Comment by swiftyspiffy** [comment]
Cool, will leave it open then. :D

**Comment by D3thw0lf** [comment]
I'm closing this now. It works. 😋👍
